cli/azd/docs/new-azd-command.md

```diff
@@ -0,0 +1,1325 @@
+# Adding New azd Commands - Comprehensive Guide
```

Comment:
We have a doc: https://github.com/Azure/azure-dev/blob/main/cli/azd/docs/adding-command.md

We might not want to have 2 docs for the same purpose?

Reply:
This one was and be more focused for AI agents to consume.